packages/preact/src/hooks/useSwatches/filterSwatches.ts

```diff
@@ -0,0 +1,26 @@
+import { SwatchField } from "../types"
+
+export function filterSwatches(swatches: SwatchField[], selectedOptions: Record<string, string>): SwatchField[] {
```

Comment:
the function name is a bit misleading, since you are not really filtering here, but adding the unavailable & selected fields